### PR TITLE
Update test_waitinglist.py

### PR DESCRIPTION
**Pull request by td15**
. Added a new class `WaitingList` that manages a queue of users using a priority queue to ensure FIFO order.
   - Methods: - `add_user(user_id)`: Adds a user to the waiting list with a timestamp. - `remove_user()`: Removes and returns the next user from the waiting list. - `peek_next_user()`: Returns the next user without removing them from the list.
   
2. Introduced a new test function `test_waiting_list()` to validate the functionality of the `WaitingList` class.
   - Tests include adding and removing users, checking the next user, and handling an empty queue case.
   - Validates that users are removed in the correct order based on their addition time.

The new functionality is integrated at the end of the existing test file, ensuring it does not interfere with the existing tests for the `WaitingListEntry` model.

## Summary by Sourcery

Implement a thread-safe WaitingList class with priority queue-based user management for the waiting list functionality

New Features:
- Introduce a WaitingList class that manages users in a queue with FIFO ordering using a thread-safe priority queue

Tests:
- Add comprehensive test cases for the WaitingList class, covering user addition, removal, and edge cases like empty queue

### CODE REVIEW

File: src/tests/api/test_waitinglist.py [hunk 1]
```diff
@@ -449,3 +451,109 @@ def test_wle_send_voucher_unavailable(token_client, organizer, event, item, wle,
...
+    for i in range(1000):
+        wl.add_user(f"LargeUser{i}")
+
+    # Validate FIFO ordering for the 1000 users
+    for i in range(1000):
+        expected = f"LargeUser{i}"
+        actual = wl.remove_user()
+        assert actual == expected, f"Test Failed: Expected {expected} but got {actual}"
+
+# Note: No direct call to test_waiting_list() here; pytest will discover and run the tests.
```
Comment (td15) [inline src/tests/api/test_waitinglist.py:559]: Next Steps:
Create a Separate File for the WaitingList Class: If you haven't already, create a separate file for the WaitingList class to keep your code organized.
Run Your Tests: Use pytest to run your tests and ensure everything works as expected.
Review and Refactor: After running the tests, review the results and refactor any code as necessary based on the outcomes.

**Comment by mariobehling** [comment]
Please open an issue first and outline the goal of your changes. Thanks
